src/goto-programs/rw_set.cpp

```diff
@@ -77,20 +80,30 @@ void rw_sett::read_write_rec(
   else if(expr.id() == "index")
   {
     assert(expr.operands().size() == 2);
-    read_write_rec(expr.op0(), r, w, suffix, guard, expr);
+    read_write_rec(expr.op0(), r, w, suffix, guard, expr, dereferenced);
   }
   else if(expr.id() == "dereference")
   {
     assert(expr.operands().size() == 1);
     read_rec(expr.op0(), guard, original_expr);
 
-    exprt tmp(expr.op0());
     expr2tc tmp_expr;
-    migrate_expr(tmp, tmp_expr);
+    migrate_expr(expr, tmp_expr);
     dereference(target, tmp_expr, ns, value_sets);
-    tmp = migrate_expr_back(tmp_expr);
+    exprt tmp = migrate_expr_back(tmp_expr);
+
+    if(
+      has_prefix(id2string(tmp.identifier()), "symex::invalid_object") ||
+      id2string(tmp.identifier()) == "")
```

Comment:
Could you please add a comment here about when this particular condition is true and what the purpose of falling back to `expr.op0()` is?

Just as a note: `dereference()` may actually return a large if-then-else expression that's not just a single symbol, so it might be worth looking out for that.